### ISSUE TITLE
I'm amazed at how bad the installation process is for this thing

### ISSUE THREAD

**Issue by pslat** [xenko3d/xenko#308]
I would just like to know how many others have tried to install and given up.

**Release Type**: Official Release
**Version**: 3.1 and 3.0
**Platform(s)**: Windows 7

**Describe the bug**
No matter what I have tried I get an error installing both the latest and 3.0 through the launcher.

**To Reproduce**
1. Try to install either version from the launcher v.3.0.4.

**Expected behavior**
A successful installation would be nice.

**Screenshots**
![xenko-error](https://user-images.githubusercontent.com/8268020/50567406-68440d80-0d3c-11e9-84f1-f3275b4dc7c8.png)

**Log and callstacks**
Error code 1 while running install package process [C:\Users\Paul\.nuget\packages\xenko.gamestudio/3.1.0.1-beta01-0318\tools\packageinstall.exe]
 at Xenko.Core.Packages.NugetStore.RunPackageInstall(String packageInstall, String arguments, ProgressReport progress) at Xenko.Core.Packages.NugetStore.OnPackageInstalled(Object sender, PackageOperationEventArgs args) at Xenko.Core.Packages.NugetStore.<InstallPackage>d50.MoveNext() --- End of stack trace from previous location where exception was thrown --- at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw() at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task) at Xenko.LauncherApp.ViewModels.PackageVersionViewModel.<>cDisplayClass550.<<Download>b_0>d.MoveNext()

**Additional context**
I've wasted a day trying to set this up. Installed the reccommended version from the launcher. I tried on my pc and laptop with similar failures. Tried installing version 3, removed 3.1 and still cannot install 3.0. I have given up and would just like to know how many others have tried to install this thing and abandoned it. My system is above min requirements, win 7 and has VS C# 2017 and Xamarin installed. The error message (error code 1 if you must know) is complete nonsense. Very annoyed that I have to stick with Unity but I'm not wasting any more time on this.

**Comment by justStand** [comment]
If I had to guess; it's a nuget issue. The package has probably not been validated yet. 

https://github.com/xenko3d/xenko/issues/295

**Comment by Kryptos-FR** [comment]
It is also possible that the nuget servers were going through some maintenance. I wouldn't recommend trying to install anything on new year's eve. It is very common for services and networks to either be crowded or under some servicing at the end of a year and beginning of the next year.

**Comment by pslat** [comment]
Thanks for the responses folks. 
justStand - validated by who or what? How do I validate the package? Reading #295 suggests some sort of network or server error but I tried again with the same result.
Kryptos-FR I have been trying since 22nd  to get this running, not only on new years eve. I tried again today (happy new year) and got the same result even though justStand was able to install the beta today. I have been reading the forums and the installation just seems to be hit or miss with everyone guessing at what might be the cause in each case. If the error message gave some meaningful message it would help. I have since tried to follow leot42 suggestion in #141 of manually installing the nuget package (downloaded  Xenko.3.0.0.6.nupkg) but the destination location seems wrong for win 7.  If this is a network problem then why doesn't the installer say that. Sure I can wait until later in the new year to try again but the whole idea was to install the software so I could try it out during the holiday season to see if I can switch over to this platform, hence my frustration with it. If I have this much trouble even getting it installed, I have to wonder if end users face similar problems when installing games based on this.

**Comment by justStand** [comment]
Validated by nuget (nothing you can do about it), but if I can get it and you can't - it's probably not what is happening here.

**Comment by pslat** [comment]
> Validated by nuget (nothing you can do about it), but if I can get it and you can't - it's probably not what is happening here.

Ok thanks anyway :(

**Comment by Kryptos-FR** [comment]
@pslat There was a potential issue with a previous version of the Launcher. But to be honest I never had any problem and I installed it in more than 3 different machines with Win7 or Win10.

**Comment by pslat** [comment]
> @pslat There was a potential issue with a previous version of the Launcher. But to be honest I never had any problem and I installed it in more than 3 different machines with Win7 or Win10.

@Kryptos-FR both machines are used for development with VS, Xamarin, Android Studio and Unity along with some other development tools so it's not worth the risk for me of trying to change anything through nuget. I may try again in the future with a cleaner machine if I ever have one.. I'm just disappointed not to be able to even try it out. I've tried installing the older v2 beta versions too at this stage with the same result so I have to give up and move on. I appreciate you taking the time to respond, thank you.

**Comment by phr00t** [comment]
Just installed 3.1 beta with the installer on Windows 10 (64 bit). No issues here. Sample application loaded and ran fine for me...

**Comment by tebjan** [comment]
works fine for me too... all packages are now validated by nuget.org. maybe try re-installing now again.

**Comment by pikachuk** [comment]
I'm posting to say I've the exact same issue on my new machine

**Comment by xen2** [comment]
BTW, sorry I didn't have a closer look at the log yet (was in vacation) but it seems to not be NuGet but the packageinstall step (which check prerequisites).
Can you check if `%UserProfile%\.nuget\packages\xenko.gamestudio\3.1.0.1-beta01-0318\tools\packageinstall.exe` actually exists?
If yes, can you run it with a debugger with all exception enabled and see if you have any exception?

**Comment by xen2** [comment]
@pslat Can you try again with latest launcher update?
It should display more info hopefully.

**Comment by aurodev** [comment]
_Didn't want to create yet another install issue since they may be related._
**System**
Windows 8.1 64bit, 8GB AMD FX 6350 6 core
Graphics Chipset
AMD Radeon HD 7700 Series
Memory Size 1024 MB
Radeon Software Version 17.4.4
.NET framework installed as of 1.21.19 .NET 4.7.2  Reg key: Release REG_DWORD 461814

For us an uninstall-Reinstall of Xenko and the launcher resulted in:

**1) Local dev copy of Xenko 3.0.0.6 that was previously building and working exiting with an invalid operation in mscorlib, built but would not run.** 

**2) Clean reinstall (multiple times) of the Xenko launcher v.3.0.5 and attempted install of 3.0.0.6 or 3.1beta  presents with:** 
`An error occurred while installing version 3.0.0.6.
Number of entries expected in End Of Central Directory does not correspond to number of entries in Central Directory. at System.IO.Compression.ZipArchive.ReadCentralDirectory() at System.IO.Compression.ZipArchive.get_Entries() at NuGet.Packaging.ZipArchiveExtensions.GetFiles(ZipArchive zipArchive) at NuGet.Protocol.LocalV2FindPackageByIdResource.GetPackageInfosCore(String id, ILogger logger) at System.Collections.Concurrent.ConcurrentDictionary2.GetOrAdd(TKey key, Func2 valueFactory) at NuGet.Protocol.LocalV2FindPackageByIdResource.GetPackageInfos(String id, SourceCacheContext cacheContext, ILogger logger) at NuGet.Protocol.LocalV2FindPackageByIdResource.GetAllVersionsAsync(String id, SourceCacheContext cacheContext, ILogger logger, CancellationToken cancellationToken) at NuGet.Commands.SourceRepositoryDependencyProvider.<GetAllVersionsAsync>d28.MoveNext() --- End of stack trace from previous location where exception was thrown --- at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw() at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task) at NuGet.Commands.SourceRepositoryDependencyProvider.<FindLibraryCoreAsync>d20.MoveNext() --- End of stack trace from previous location where exception was thrown --- at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw() at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task) at NuGet.Commands.SourceRepositoryDependencyProvider.<>cDisplayClass19_0.<<FindLibraryAsync>b0>d.MoveNext() --- End of stack trace from previous location where exception was thrown --- at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw() at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task) at NuGet.Commands.SourceRepositoryDependencyProvider.<FindLibraryAsync>d19.MoveNext() --- End of stack trace from previous location where exception was thrown --- at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw() at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task) at NuGet.DependencyResolver.ResolverUtility.<>cDisplayClass7_1.<<FindLibraryFromSourcesAsync>b0>d.MoveNext() --- End of stack trace from previous location where exception was thrown --- at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw() at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task) at NuGet.DependencyResolver.ResolverUtility.<FindLibraryFromSourcesAsync>d7.MoveNext() --- End of stack trace from previous location where exception was thrown --- at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
`



We've installed and built Xenko many times and worked through it's idiosyncrasies each time.
Xenko seems to work well, until it doesn't. 


Of course all of this can be fixed but serves to illustrate just how fragile Xenko is.

**Comment by xen2** [comment]
In reference to original issue (packageinstall.exe failing), I got a more detailed log from Discord:
![image](https://user-images.githubusercontent.com/527565/51544239-cfb82d80-1ea2-11e9-87d4-b774c9f5368d.png)
It looks like UAC was denied.

@pslat There was no UAC popup whatsoever?
What is your current UAC setting?
What happens if you run `"%PROGRAMFILES(x86)%\Microsoft Visual Studio\Installer\vs_installer.exe"` manually from a command line, does UAC pops up?

@aurodev thanks for the report. It looks like a completely different issue so might be worth opening a new github issue, thanks.

**Comment by xen2** [comment]
I did a bunch of improvements in Xenko 3.0.0.7 and latest 3.1 beta.
Can you give it another try please, and reopen/comment if you still have issues?

**Comment by pslat** [comment]
Ok sorry folks for the delay. My AV was detecting xenko as malware and quarantening some of the files. I have reported this to Avira and they have confirmed a false positive. I expect the latest update to correct this. I had tried to install with the AV turned off but still wasn't successful. %UserProfile%\.nuget\packages\xenko.gamestudio\3.1.0.1-beta01-0318\tools\packageinstall.exe does not exist on my system except at install time briefly. There are a lot of Xenko folders in there including gamestudio but that one is empty again after the error. @xen2 I had already started looking into the nuget issue as a possibility but I'm afraid I had got to the stage of removing all folders before I got your message. I don't know which one did it but I was finally able to get xenko 3.0 installed on the desktop machine. I do not have a http proxy installed btw. Unfortunately I tried the fps demo and had another round of errors to get through before I got it running and found that it ran like a one legged dog on my machine.  I spent a lot of time then trying to get an Android demo project to compile, searching the forums and carrying out the suggestions but again got to the stage where I had enough of this. I do not like to give up on these things but I expect the performance would be even worse on the laptop if it ran at all. Below is the latest install attempt fyi anyway. Please note I do not have a D: drive on my system so I don't know where it's getting that from unless the installer created a virtual drive to work from. @xen2 I do get the UAC prompt and allow the installation to continue. I think someone else posted that rather than me. I get a second UAC message for prerequisites and allow that also. Perhaps it is still the antivirus but as I managed to get it installed on the other pc I don't think so:( It is clear to me now that my issues are the exception to the rule and most others have not experienced these problems so I'm happy to have the thread closed. I regret the title for the thread now..Perhaps I'll try out Xenko again in the future best of luck to the developers who have obviously out a lot of work into this project.

 For reference, here is the last error message:
Error code 1 while running install package process [C:\Users\Paul\.nuget\packages\xenko.gamestudio/3.1.0.1-beta01-0406\tools\packageinstall.exe]
Error: System.UnauthorizedAccessException: Retrieving the COM class factory for component with CLSID {177F0C4A-1CD3-4DE7-A32C-71DBBB9FA36D} failed due to the following error: 80070005 Access is denied. (Exception from HRESULT: 0x80070005 (E_ACCESSDENIED)). at Xenko.Core.VisualStudio.VisualStudioVersions.BuildIDEInfos() in D:\BuildAgent\work\8cafb933a54b01d0\sources\core\Xenko.Core.Design\VisualStudio\VisualStudioVersions.cs:line 127 at System.Lazy1.CreateValue()
   at System.Lazy1.LazyInitValue() at Xenko.Core.VisualStudio.VisualStudioVersions.get_AvailableVisualStudioInstances() in D:\BuildAgent\work\8cafb933a54b01d0\sources\core\Xenko.Core.Design\VisualStudio\VisualStudioVersions.cs:line 94 at Xenko.PackageInstall.Program.CheckVisualStudioAndBuildTools() in D:\BuildAgent\work\8cafb933a54b01d0\sources\tools\Xenko.PackageInstall\Program.cs:line 102 at Xenko.PackageInstall.Program.Main(String[] args) in D:\BuildAgent\work\8cafb933a54b01d0\sources\tools\Xenko.PackageInstall\Program.cs:line 51
 at Xenko.Core.Packages.NugetStore.RunPackageInstall(String packageInstall, String arguments, ProgressReport progress) at Xenko.Core.Packages.NugetStore.OnPackageInstalled(Object sender, PackageOperationEventArgs args) at Xenko.Core.Packages.NugetStore.<InstallPackage>d50.MoveNext() --- End of stack trace from previous location where exception was thrown --- at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw() at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task) at Xenko.LauncherApp.ViewModels.PackageVersionViewModel.<>cDisplayClass550.<<Download>b_0>d.MoveNext()

**Comment by xen2** [comment]
Thanks for the update.
Files appearing shortly and disappearing really sound like an antivirus messing. That's strange it was still happening even though you disabled it.

**Comment by pslat** [comment]
@xen2 the files are removed after the install fails. I've just copied the folder  .nuget\packages\xenko.gamestudio\3.1.0.1-beta01-0406 during another attempted install (as soon as it appeared) and was able to retain the copy of the folder with contents including packageinstaller.exe. If it was the antivirus removing it then it would have also detected the folder copy and deleted that. Also the av doesn't delete folders, just the suspected files within folders. As I had now a copy of the 3.1.0.1-beta01-0406 folder I tried running packageinstaller.exe from there and got the same error in the command prompt. It was not the antivirus removing the folder, it has to be the installer or nuget cleaning up after itself when the installation fails.
